logger/redact.go

```diff
@@ -86,6 +121,29 @@ func RedactJSON(json, replacement string, keys [][]string) string {
 	return string(out)
 }
 
+func redactRecursive(obj *fastjson.Value, keys []string, replacementValue *fastjson.Value) {
```

Comment:
do you think it is better to rename this to `redactArrayRecursive`? (since it is used only for masking array)

Reply:
Done in 24788db.